subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -985,6 +985,51 @@ static int client_broker_init(struct mqtt_client *const client)
 	return err;
 }
 
+static int connect_client(struct aws_iot_config *const config)
+{
+	int err;
+
+	err = client_broker_init(&client);
+	if (err) {
+		LOG_ERR("client_broker_init, error: %d", err);
+		return err;
+	}
+
+	err = mqtt_connect(&client);
+	err = connect_error_translate(err);
+	if (err) {
+		LOG_ERR("mqtt_connect, error: %d", err);
```

Comment:
Hm, this is in reality the return code from `connect_error_translate`, which is different from the one returned by `mqtt_connect`. I think we should print the return code from `mqtt_connect` too for debugging. And wrap the translation in an `if (err) { ... }` and do it only if `mqtt_connect` fails